### PR TITLE
encrypt api key with kms

### PR DESCRIPTION
**Pull request by guangrui-cloudnatix**

### CODE REVIEW

File: deployments/server/templates/configmap.yaml [hunk 1]
```diff
@@ -35,3 +35,6 @@ data:
       rbacInternalServerAddr: {{ .Values.global.auth.rbacInternalServerAddr }}
     usageSender:
       {{- toYaml .Values.global.usageSender | nindent 6 }}
+    kms:
+      enable: {{ .Values.global.kms.enable }}
```
Comment (kkaneda) [inline deployments/server/templates/configmap.yaml:39]: Question: Do we need `region` as well?

**Review by kkaneda — COMMENTED** [review]
Cool!

File: server/cmd/run.go [hunk 2]
```diff
@@ -107,7 +109,32 @@ func run(ctx context.Context, c *config.Config) error {
...
+		)
+		if err != nil {
+			return err
+		}
+		dk, err := st.GetDataKey(ctx, kmsClient)
+		if err != nil {
+			if !errors.Is(err, gorm.ErrRecordNotFound) {
+				return err
+			}
+			dk, err = st.CreateDataKey(ctx, kmsClient)
```
Comment (kkaneda) [inline server/cmd/run.go:129]: How about adding a logging message here? e.g., "Creating a new data key"

File: deployments/server/templates/configmap.yaml (hunk 1 above)
Comment (guangrui-cloudnatix) [reply deployments/server/templates/configmap.yaml]: Yes, added

File: server/internal/store/api_key.go [hunk 3]
```diff
@@ -18,7 +18,10 @@ type APIKey struct {
 	ProjectID      string
 	UserID         string `gorm:"uniqueIndex:idx_api_key_name_user_id"`
 
+	// Secret is set when kms encryption is disabled.
 	Secret string
+	// EncryptedAPIKey is encrypted by data key, and it is set when kms encryption is enabled.
+	EncryptedAPIKey []byte
```
Comment (kkaneda) [inline server/internal/store/api_key.go:24]: How about `EncryptedSecret`?

**Review by kkaneda — APPROVED** [review]
LG!

**Pull request merged**